### PR TITLE
Settings modal

### PR DESCRIPTION
**Pull request by SeanMcCarren**
- [x] Settings modal
- [x] UI Theming
- [x] Fonts can be changed
- [ ] More themes?
- [ ] Test cases (I think SettingsModal.vue can be unit tested (and the event bus?) but I don't know how)

To create themes yourself is not very trivial... But here are the steps anyway
- Clone file light.scss in src/assets/sass/abstracts and make this your own theme definition! Do not remove any lines here, just change the values.
- Add reference in src/sass/main
- Add reference and generateThemeMapping in src/assets/sass/colors
- Go to SettingsModal.vue and add map name to styles list

Additionally, you may want to use new fonts in your style, in which case you should:
- Add font pack (in woff2 or the likes) to src/assets/fonts
- Define fonts in src/assets/sass/base/fonts

### CODE REVIEW

File: src/store/store.js [hunk 1]
```diff
@@ -60,9 +65,43 @@ export default {
...
+    setZoom: function(state, factor) {
+      state.settings.zoom = factor;
+      const wf = require('electron').webFrame;
+      wf.setZoomFactor(factor);
+
+      updateConfiguration(remote, state.settings);
+    },
+    setTheme: function(state, theme) {
+      state.settings.theme = theme;
+      document.getElementsByTagName('HTML')[0].setAttribute('class', theme);
```
Comment (davidot) [inline src/store/store.js:102]: This is a bit more complicated then it needs to be
```suggestion
      document.documentElement.className = theme;
```

File: src/store/store.js [hunk 2]
```diff
@@ -60,9 +65,43 @@ export default {
...
+    setConfig: function(state, result) {
+      if (result.hasOwnProperty('theme')) {
+        state.settings.theme = result['theme'];
+      }
+
+      // Conformance. Must be 'light' or 'dark'. Default is light
+      if (state.settings.theme == null || state.settings.theme === '') {
+        state.settings.theme = 'light';
+      }
+      document.getElementsByTagName('HTML')[0]
```
Comment (davidot) [inline src/store/store.js:77]: Same here `document.documentElement.className = theme;`

File: src/index/components/HomepageButton.vue [hunk 3]
```diff
@@ -103,6 +104,7 @@ export default {
    * Ensure text is vertically centered.
    */
   &-text {
+    @include theme(color, color-text-in-primary);
```
Comment (davidot) [inline src/index/components/HomepageButton.vue:107]: Is now white on white should be `color-primary`?

File: src/editpage/components/assistance/SettingsModal.vue [hunk 4]
```diff
@@ -0,0 +1,232 @@
...
+  },
+};
+</script>
+
+<style lang="scss">
+  @import '../../../assets/sass/pages/edit';
+
+  /* The Modal (background) */
+  #settingsModal {
+    background-color: rgba(255,255,255,0.5); /* Partly opaque background */
```
Comment (davidot) [inline src/editpage/components/assistance/SettingsModal.vue:145]: Maybe a slightly darker color would make it more clear that the content behind is blocked.
Also maybe the modal could close if you click on the background.

File: src/editpage/components/assistance/SettingsModal.vue [hunk 5]
```diff
@@ -0,0 +1,232 @@
...
+    changeTheme: function(theme) {
+      if (this.styles.includes(theme)) {
+        this.$store.commit('setTheme', theme);
+      }
+    },
+
+    updateConfigurationString: function() {
+      const libs = this.$store.state.libraries;
+      const settings = this.$store.state.settings;
+      document.getElementById('settingsOverview').innerHTML =
```
Comment (davidot) [inline src/editpage/components/assistance/SettingsModal.vue:124]: Why are we creating this html manually?

File: src/editpage/components/assistance/SettingsModal.vue [hunk 6]
```diff
@@ -0,0 +1,232 @@
...
+<script>
+import Vue from 'vue';
+
+export default {
+  name: 'SettingsModal',
+  components: {},
+  props: {},
+  data: function() {
+    return {
+      eventBus: new Vue(),
```
Comment (davidot) [inline src/editpage/components/assistance/SettingsModal.vue:69]: Within this component I don't think we need an entire eventbus

**Review by davidot — CHANGES_REQUESTED** [review]
Small improvements

File: src/io/configurationio.js [hunk 7]
```diff
@@ -5,8 +5,11 @@ const defaultConfigData = {
   sertopPath: '',
   serapiVersion: '',
   libraryVersion: '',
+  zoom: '',
+  theme: '',
 };
-const possibleKeys = ['sertopPath', 'serapiVersion', 'libraryVersion'];
+const possibleKeys = [
```
Comment (davidot) [inline src/io/configurationio.js:11]: This is annoying to have to add for every setting we should probably change this.
Out of scope for this PR though.

File: src/editpage/components/assistance/SettingsModal.vue (hunk 6 above)
Comment (SeanMcCarren) [reply src/editpage/components/assistance/SettingsModal.vue:69]: I'm not sure, but is there a better way to enable it from Sidebar.vue in that case? Now, we use that event bus

File: src/editpage/components/assistance/SettingsModal.vue (hunk 5 above)
Comment (SeanMcCarren) [reply src/editpage/components/assistance/SettingsModal.vue:124]: I agree it's quite ugly. Any suggestions? I want to use the v-for but then libs has to be in the data somehow...

**Review by davidot — APPROVED** [review]

**Pull request merged**